packages/web/src/pages/user/[uid].tsx

```diff
@@ -38,22 +40,37 @@ const UserProfilePage: NextPage = () => {
     fetchUser();
   }, [uid]);
 
+  React.useEffect(() => {
+    const checkUser = async () => {
+      const res = await fetch('/api/users/me');
+      if (res.status === 200) {
+        const data = await res.json();
+
+        if (user && data.id == user.id) {
+          setIsCurrentUser(true);
+        }
+      }
```

Comment:
Wrap this in a `try catch` and then add an appropriate error state to the UI

Reply:
If the check for the current user throws an error, should the queried user's profile still be shown or should the error message by the only thing displayed?